### DISCUSSION TITLE
Self-closing custom components

### DISCUSSION THREAD

**Discussion by viicslen** [GrapesJS/grapesjs#5205] [Q&A] [answered]
I have the following custom component that I want to make self closing.

Current Output
```html
<body>
  <variable-item group="item">
  </variable-item>Hello World
</body>
```

Desired Output
```html
<body>
  <variable-item group="item" />Hello World
</body>
```

I have tried modifying the code in the `toHTML`  function but when done like this importing the desired html back again results in the following :

```html
<body>
  <variable-item group="item">Hello World
  </variable-item>
</body>
```

**Comment by artf** [comment, marked as answer]
This is not related to GrapesJS, [you can't have custom void/self-closing elements](https://github.com/WICG/webcomponents/issues/624)
